File: anti-thief-new/app/src/main/java/com/emopass/antitheftalarm/adapter/OnBoardingViewPagerAdapter.java
package com.emopass.antitheftalarm.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.emopass.antitheftalarm.R;
import com.emopass.antitheftalarm.databinding.OnboardingViewBinding;
import com.emopass.antitheftalarm.model.OnBoardingModel;
import java.util.ArrayList;
import java.util.List;

public class OnBoardingViewPagerAdapter extends RecyclerView.Adapter<OnBoardingViewPagerAdapter.ViewHolder> {
    private Context context;
    private List<OnBoardingModel> onBoardingModelList = new ArrayList<>();
    public OnBoardingViewPagerAdapter(Context context) {
        this.context = context;
        onBoardingModelList.add(new OnBoardingModel(context.getString(R.string.onboarding1_title), R.drawable.front_onboarding_screen_1));
        onBoardingModelList.add(new OnBoardingModel(context.getString(R.string.onboarding2_title), R.drawable.front_onboarding_screen_2));
        onBoardingModelList.add(new OnBoardingModel(context.getString(R.string.onboarding3_title), R.drawable.front_onboarding_screen_3));

    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        OnboardingViewBinding binding = OnboardingViewBinding.inflate(
                LayoutInflater.from(parent.getContext()),
                parent,
                false
        );
        return new ViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.onBind(position);
    }

    @Override
    public int getItemCount() {
        return onBoardingModelList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private OnboardingViewBinding binding;
        public ViewHolder(OnboardingViewBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public OnboardingViewBinding getBinding() {
            return binding;
        }

        public void onBind(int position) {
            OnBoardingModel model = onBoardingModelList.get(position);
            if(position == 0) {
                binding.imgOnBoarding.setScaleType(ImageView.ScaleType.FIT_START);
            }
            binding.imgOnBoarding.setImageResource(model.getImageRes());
            binding.txtTitle.setText(model.getLabel());
        }
    }
}

File: Wallpaper/app/src/main/java/com/parallax/hdvideo/wallpapers/ui/onboarding/OnBoardingViewPagerAdapter.kt
package com.parallax.hdvideo.wallpapers.ui.onboarding

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.parallax.hdvideo.wallpapers.R
import com.parallax.hdvideo.wallpapers.data.model.OnBoardingPageData
import com.parallax.hdvideo.wallpapers.databinding.OnboardingViewBinding


class OnBoardingViewPagerAdapter(val context: Context) :
    RecyclerView.Adapter<OnBoardingViewPagerAdapter.PageViewHolder>() {

    private val onBoardingPageDataLists = listOf(
        OnBoardingPageData(
            context.getString(R.string.onboarding_title_1),
            R.drawable.front_onboarding_screen_1, R.drawable.bg_onboarding_screen_1
        ),
        OnBoardingPageData(
            context.getString(R.string.onboarding_title_2),
            R.drawable.front_onboarding_screen_2, R.drawable.bg_onboarding_screen_2
        ),
        OnBoardingPageData(
            context.getString(R.string.onboarding_title_3),
            R.drawable.front_onboarding_screen_3, R.drawable.bg_onboarding_screen_3
        )
    )

    inner class PageViewHolder(private val binding: OnboardingViewBinding) :
        RecyclerView.ViewHolder(binding.root) {
        fun onBind(position: Int) {
            val onBoardingPageData = onBoardingPageDataLists[position]
            if(position == 0) {
                binding.imgOnBoarding.scaleType = ImageView.ScaleType.FIT_START
            }
            binding.imgOnBoarding.setImageResource(onBoardingPageData.imageResId)
            //binding.background.setImageResource(onBoardingModel.background)
            binding.txtDescription.visibility = View.INVISIBLE
            binding.txtTitle.text = onBoardingPageData.label
        }
    }

    override fun getItemCount(): Int = 3

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PageViewHolder {
        return PageViewHolder(
            OnboardingViewBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        )
    }

    override fun onBindViewHolder(holder: PageViewHolder, position: Int) {
        holder.onBind(position)
    }
}